From f4e316ef4d8d8c9a558ef5bfa74e25303c46a985 Mon Sep 17 00:00:00 2001
From: Srinivas Pandruvada <srinivas.pandruvada@linux.intel.com>
Date: Mon, 25 Jan 2016 14:27:42 -0800
Subject: [PATCH 203/203] Add white list of the cpu ids

To avoid surprises add all whiltelist of Intel client CPUs which is upto
date till 2016.
User can overide this by either command line parameter
--ignore-cpuid-check or define a thermal-conf.xml. So this still
can be used on AMD or other unsupported models.
---
 src/thd_engine.cpp | 27 ++++++++++++++++++++++-----
 1 file changed, 22 insertions(+), 5 deletions(-)

diff --git a/src/thd_engine.cpp b/src/thd_engine.cpp
index 9b9b014..d75d268 100644
--- a/src/thd_engine.cpp
+++ b/src/thd_engine.cpp
@@ -169,9 +169,19 @@ int cthd_engine::thd_engine_start(bool ignore_cpuid_check) {
 	if (ignore_cpuid_check) {
 		thd_log_debug("Ignore CPU ID check for MSRs \n");
 		proc_list_matched = true;
-	} else
+	} else {
 		check_cpu_id();
 
+		if (!proc_list_matched) {
+			if ((parser_init() == THD_SUCCESS) && parser.platform_matched()) {
+				thd_log_warn("Unsupported cpu model, using thermal-conf.xml only \n");
+			} else {
+				thd_log_warn("Unsupported cpu model, use thermal-conf.xml file or run with --ignore-cpuid-check \n");
+				return THD_FATAL_ERROR;
+			}
+		}
+	}
+
 	check_for_rt_kernel();
 
 	// Pipe is used for communication between two processes
@@ -576,11 +586,18 @@ void cthd_engine::thd_engine_reload_zones() {
 }
 
 // Add any tested platform ids in this table
-static supported_ids_t id_table[] = { { 6, 0x2a }, // Sandybridge
-		{ 6, 0x2d }, // Sandybridge
+static supported_ids_t id_table[] = {
+		{ 6, 0x2a }, // Sandybridge
 		{ 6, 0x3a }, // IvyBridge
-		{ 6, 0x3c }, { 6, 0x3e }, { 6, 0x3f }, { 6, 0x45 }, // Haswell ULT */
-		{ 6, 0x46 }, // Haswell ULT */
+		{ 6, 0x3c }, // Haswell
+		{ 6, 0x45 }, // Haswell ULT
+		{ 6, 0x46 }, // Haswell ULT
+		{ 6, 0x3d }, // Broadwell
+		{ 6, 0x37 }, // Valleyview BYT
+		{ 6, 0x4c }, // Brasewell
+		{ 6, 0x4e }, // skylake
+		{ 6, 0x5e }, // skylake
+		{ 6, 0x5c }, // Broxton
 
 		{ 0, 0 } // Last Invalid entry
 };
-- 
2.7.0.rc3

